### PR TITLE
[ESIMD] Add bfloat16 test cases.

### PR DESCRIPTION
**Pull request by kbobrovs**
Complementary compiler patch: https://github.com/intel/llvm/pull/6664

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/api/unary_ops_heavy.cpp [hunk 1]
```diff
@@ -172,14 +174,20 @@ int main(void) {
...
+  auto signed_ops = esimd_test::OpSeq<UnOp, UnOp::minus, UnOp::plus>{};
+  passed &= test<char, 7>(signed_ops, q);
+  passed &= test<short, 7>(signed_ops, q);
+  passed &= test<int, 16>(signed_ops, q);
+  passed &= test<int64_t, 16>(signed_ops, q);
+  passed &= test<half, 16>(signed_ops, q);
+  passed &= test<float, 16>(signed_ops, q);
+  passed &= test<double, 16>(signed_ops, q);
+
+#ifdef USE_BF16
```
Comment (v-klochkov) [inline SYCL/ESIMD/api/unary_ops_heavy.cpp:187]: The obvious downside of this approach is the duplication of checks/tests.
I.e. on PVC the lines 178-184 started twice: 1st time as part of unary_ops_heavy.cpp test, and then (the same code + line 189) as part of unary_ops_heavy_pvc.cpp

Comment (kbobrovs) [reply SYCL/ESIMD/api/unary_ops_heavy.cpp:187]: Right, good catch! I will add proper test selection.

**Review by v-klochkov — APPROVED** [review]

**Comment by kbobrovs** [comment]
Linux all passed (Jenkins/pre-ci-linux did not refresh icon for some reason), Windows - usual fp64 related failures unrelated to this patch. Merging.

**Pull request merged**